basic-parser.js

```diff
@@ -0,0 +1,378 @@
+import { parse } from './parser/dsl.js'
+import { readFileSync } from 'fs'
+
+/**
+ * @pineapple_define Parser
+ */
+export function InternalTests () {
+  return {
+    voidTest: `
+          /** 
+           * @test void
+           */
+          `,
+    addTest: `
+          /**
+           * @test 1, 
+           * 2 returns 3
+           */
+          export function add (a, b) {
+              return a + b
+          }
+      `,
+    moduleExports: `
+          module.exports = { 
+              "hello": age, name 
+          }
+      `,
+    typicalExport: `
+          export { X, Y as Z }
+          export { Two }
+    `,
+    exports: `
+      exports.X = XYZ;
+      exports.Y = Y;
+    `,
+    quoteIssue: `
+          function a (x = /helloWorld\\//) {
+
+          }
+
+          function b () {
+            return "hello{" + 'wrld{\\''
+          }
+
+          function c () {
+
+          }
+    `,
+    this: readFileSync('./basic-parser.js').toString(),
+    mathCjs: readFileSync('./test/math.cjs').toString(),
+    passwordModule: readFileSync('./test/password_module.js').toString(),
+    setupTest: readFileSync('./test/setup.test.js').toString()
+  }
+}
+
+/**
+ * This function removes all of the strings, Regexes and contents in between braces.
+ * This is done to make it simpler to run a regex to fetch out classes,
+ * functions and variable declarations.
+ * @test "function a () { function b () {} return b }"
+ * @test 'function a () { function b () {} return b } function b () { }'
+ * @test 'z + { { a } + } { b } + c'
+ * @test #Parser.this
+ * @test #Parser.quoteIssue
+ * @test #Parser.passwordModule
+ * @test #Parser.setupTest
+ * @param {string} code
+ */
+export function strip (code) {
```

Comment:
Honestly speaking, if someone else handed me this function, I would reject it in a PR. 

I'd say that this would be difficult to read & maintain, and I'd try to refactor it. 

There are a few approaches that could be taken to make this simpler:
- Define a Peggy grammar that handles all of the extraction. (A little tricky to write -- not impossible)
- Define simpler Peggy grammars to strip out regexes and strings.
- Just make this loop a bit cleaner.

I may go ahead and merge this in as-is since we're still in the "get it working" stages of development, and the speed boost is rather decent.